clients/banking/src/components/MerchantProfilePaymentLinkArea.tsx

```diff
@@ -0,0 +1,292 @@
+            color="current"
+            onPress={() =>
+              Router.push("AccountMerchantsProfilePaymentLinkList", {
+                new: "true",
+                accountMembershipId,
+                merchantProfileId,
+              })
+            }
+          >
+            {t("merchantProfile.paymentLink.button.new")}
+          </LakeButton>
+        </Box>
+      )}
+
+      <Box
+        direction="row"
+        alignItems="center"
+        style={[styles.filters, large && styles.filtersLarge]}
+      >
+        {large && (
+          <>
+            <LakeButton
+              size="small"
+              icon="add-circle-filled"
+              color="current"
+              onPress={() =>
+                Router.push("AccountMerchantsProfilePaymentLinkList", {
+                  new: "true",
+                  accountMembershipId,
+                  merchantProfileId,
+                })
+              }
+            >
+              {t("merchantProfile.paymentLink.button.new")}
+            </LakeButton>
+
+            <Space width={12} />
+          </>
+        )}
+
+        <LakeButton
+          ariaLabel={t("common.refresh")}
+          mode="secondary"
+          size="small"
+          icon="arrow-counterclockwise-filled"
+          loading={isRefreshing}
+          onPress={() => {
+            setIsRefreshing(true);
+            reload().tap(() => setIsRefreshing(false));
+          }}
+        />
+
+        <Fill minWidth={16} />
+
+        <Box direction="row" alignItems="center" justifyContent="end" style={styles.endFilters}>
+          <Toggle
+            mode={large ? "desktop" : "mobile"}
+            value={params.status === "Active"}
+            onToggle={status =>
+              Router.push("AccountMerchantsProfilePaymentLinkList", {
+                accountMembershipId,
+                merchantProfileId,
+                status: status ? "Active" : "Archived",
+              })
+            }
+            onLabel={t("merchantProfile.list.Active")}
+            offLabel={t("merchantProfile.list.Inactive")}
+          />
+
+          <Fill minWidth={16} />
+
+          <LakeSearchField
+            initialValue={search ?? ""}
+            placeholder={t("common.search")}
+            onChangeText={search => {
+              Router.replace("AccountMerchantsProfilePaymentLinkList", {
+                accountMembershipId,
+                merchantProfileId,
+                ...params,
+                search,
+              });
+            }}
+          />
+        </Box>
+      </Box>
+
+      <Space height={24} />
+
+      {match(data)
+        .with(AsyncData.P.NotAsked, AsyncData.P.Loading, () => (
+          <PlainListViewPlaceholder count={5} groupHeaderHeight={48} rowHeight={56} />
+        ))
+        .with(AsyncData.P.Done(Result.P.Error(P.select())), error => <ErrorView error={error} />)
+        .with(AsyncData.P.Done(Result.P.Ok(P.select())), ({ merchantProfile }) => (
```

Comment:
It could be good to transform `data` to make sure a `merchantProfile` which is `null | undefined` result in an error here, instead of the nullability checks (I don't think we want to display something else than an error if the API returns nothing)